frontend-admin/src/app/components/mnoe-api/admin/current-user.svc.coffee

```diff
@@ -28,13 +28,32 @@
     )
 
   @skipIfNotAdmin = () ->
-    if _self.user.admin_role? && _self.user.admin_role == 'admin'
-      return $q.resolve()
-    else
-      $timeout(->
-        # Runs after the authentication promise has been rejected.
-        $state.go('dashboard.home')
-      )
-      $q.reject()
+    deferred = $q.defer()
+    _self.getAdminRole().then(->
+      if _self.user.admin_role? && _self.user.admin_role == 'admin'
+        return deferred.resolve()
+      else
+        $timeout(->
+          # Runs after the authentication promise has been rejected.
+          $state.go('dashboard.home')
+        )
+        deferred.reject()
+    )
+    return deferred
+
+  @logoutIfNoOrganisation = ->
+    userPromise.then(
+      (response) ->
+        # Redirect to dashboard if the user has at least one organization
+        if response.data.organizations? && response.data.organizations.length > 0
+          $window.location.assign("/dashboard/")
+        # Logout if the user has no organization
+        else
+          $window.location.assign("/dashboard/#/logout")
+    )
+
+  @updateAdminRole = ->
+    userPromise = null
+    _self.skipIfNotAdmin()
```

Comment:
I think you should then call `_self.getUser()` now here to refresh the user. Call this function `refreshUser()`